cli_test.{test_verbose,test_verbose_and_single_child} fail on some architectures due to signal number assertions

Signals have different numbers on different architectures, and these tests assert that SIGCHLD is 17 (but on mips, it is 18!).

Should be easy to fix by just formatting in `signal.SIGCHLD` to the assertion string.

The Debian mips build for dumb-init failed: https://buildd.debian.org/status/fetch.php?pkg=dumb-init&arch=mips&ver=1.1.2-1&stamp=1469941155&file=log

So did the mipsel build: https://buildd.debian.org/status/fetch.php?pkg=dumb-init&arch=mipsel&ver=1.1.2-1&stamp=1469941005&file=log

```
=================================== FAILURES ===================================
_______________________________ test_verbose[-v] _______________________________

flag = '-v'

    @pytest.mark.parametrize('flag', ['-v', '--verbose'])
    def test_verbose(flag):
        """dumb-init should print debug output when asked to."""
        proc = Popen(('dumb-init', flag, 'echo', 'oh,', 'hi'), stdout=PIPE, stderr=PIPE)
        stdout, stderr = proc.communicate()
        assert proc.returncode == 0
        assert stdout == b'oh, hi\n'
    
        # child/parent race to print output after the fork(), can't guarantee exact order
        assert re.search(b'(^|\n)\[dumb-init\] setsid complete\.\n', stderr), stderr  # child
>       assert re.search(  # parent
            (
                b'(^|\n)\[dumb-init\] Child spawned with PID [0-9]+\.\n'
                b'.*'  # child might print here
                b'\[dumb-init\] Received signal 17\.\n'
                b'\[dumb-init\] A child with PID [0-9]+ exited with exit status 0.\n'
                b'\[dumb-init\] Forwarded signal 15 to children\.\n'
                b'\[dumb-init\] Child exited with status 0\. Goodbye\.\n$'
            ),
            stderr,
            re.DOTALL,
        ), stderr
E       AssertionError: [dumb-init] setsid complete.
E         [dumb-init] Child spawned with PID 27262.
E         [dumb-init] Received signal 18.
E         [dumb-init] A child with PID 27262 exited with exit status 0.
E         [dumb-init] Forwarded signal 15 to children.
E         [dumb-init] Child exited with status 0. Goodbye.
E         
E       assert None
E        +  where None = <function search at 0x77249270>('(^|\n)\\[dumb-init\\] Child spawned with PID [0-9]+\\.\n.*\\[dumb-init\\] Received signal 17\\.\n\\[dumb-init\\] A ch...us 0.\n\\[dumb-init\\] Forwarded signal 15 to children\\.\n\\[dumb-init\\] Child exited with status 0\\. Goodbye\\.\n$', '[dumb-init] setsid complete.\n[dumb-init] Child spawned with PID 27262.\n[dumb-init] Received signal 18.\n[dumb-init]... with exit status 0.\n[dumb-init] Forwarded signal 15 to children.\n[dumb-init] Child exited with status 0. Goodbye.\n', 16)
E        +    where <function search at 0x77249270> = re.search
E        +    and   16 = re.DOTALL
```

--- a/testing/__init__.py
+++ b/testing/__init__.py
@@ -1,3 +1,4 @@
+import errno
 import os
 import re
 import signal
@@ -92,3 +93,12 @@ def sleep_until(fn, timeout=1.5):
             break
         time.sleep(interval)
         so_far += interval
+
+
+def kill_if_alive(pid, signum=signal.SIGKILL):
+    """Kill a process, ignoring "no such process" errors."""
+    try:
+        os.kill(pid, signum)
+    except OSError as ex:
+        if ex.errno != errno.ESRCH:  # No such process
+            raise

--- a/tests/child_processes_test.py
+++ b/tests/child_processes_test.py
@@ -8,6 +8,7 @@ from subprocess import Popen
 import pytest
 
 from testing import is_alive
+from testing import kill_if_alive
 from testing import pid_tree
 from testing import sleep_until
 
@@ -60,7 +61,7 @@ def test_no_setsid_doesnt_signal_entire_group():
     sleep_until(assert_four_living_pids)
 
     for pid in living_pids(pids):
-        os.kill(pid, signal.SIGKILL)
+        kill_if_alive(pid)
 
 
 def spawn_process_which_dies_with_children():

--- a/tests/cli_test.py
+++ b/tests/cli_test.py
@@ -1,5 +1,6 @@
 """Sanity checks for command-line options."""
 import re
+import signal
 from subprocess import PIPE
 from subprocess import Popen
 
@@ -87,13 +88,13 @@ def test_verbose(flag):
     assert re.search(b'(^|\n)\[dumb-init\] setsid complete\.\n', stderr), stderr  # child
     assert re.search(  # parent
         (
-            b'(^|\n)\[dumb-init\] Child spawned with PID [0-9]+\.\n'
-            b'.*'  # child might print here
-            b'\[dumb-init\] Received signal 17\.\n'
-            b'\[dumb-init\] A child with PID [0-9]+ exited with exit status 0.\n'
-            b'\[dumb-init\] Forwarded signal 15 to children\.\n'
-            b'\[dumb-init\] Child exited with status 0\. Goodbye\.\n$'
-        ),
+            '(^|\n)\[dumb-init\] Child spawned with PID [0-9]+\.\n'
+            '.*'  # child might print here
+            '\[dumb-init\] Received signal {signal.SIGCHLD}\.\n'
+            '\[dumb-init\] A child with PID [0-9]+ exited with exit status 0.\n'
+            '\[dumb-init\] Forwarded signal 15 to children\.\n'
+            '\[dumb-init\] Child exited with status 0\. Goodbye\.\n$'
+        ).format(signal=signal).encode('utf8'),
         stderr,
         re.DOTALL,
     ), stderr
@@ -109,12 +110,12 @@ def test_verbose_and_single_child(flag1, flag2):
     assert stdout == b'oh, hi\n'
     assert re.match(
         (
-            b'^\[dumb-init\] Child spawned with PID [0-9]+\.\n'
-            b'\[dumb-init\] Received signal 17\.\n'
-            b'\[dumb-init\] A child with PID [0-9]+ exited with exit status 0.\n'
-            b'\[dumb-init\] Forwarded signal 15 to children\.\n'
-            b'\[dumb-init\] Child exited with status 0\. Goodbye\.\n$'
-        ),
+            '^\[dumb-init\] Child spawned with PID [0-9]+\.\n'
+            '\[dumb-init\] Received signal {signal.SIGCHLD}\.\n'
+            '\[dumb-init\] A child with PID [0-9]+ exited with exit status 0.\n'
+            '\[dumb-init\] Forwarded signal 15 to children\.\n'
+            '\[dumb-init\] Child exited with status 0\. Goodbye\.\n$'
+        ).format(signal=signal).encode('utf8'),
         stderr,
     )
 

